collisions_main.py

```diff
@@ -0,0 +1,246 @@
+        num_y_sections = int(((boundary[3] - boundary[2])
+                              / base_section_dim) + 1)
+
+                  
+    # New object radius slider
+    if click_x >= RAD_SLIDER_X_MIN and click_x <= RAD_SLIDER_X_MAX:
+        if (click_y * RAD_SLIDER_MULT != max_rad and
+                click_y * RAD_SLIDER_MULT > 0):
+            max_rad = math.ceil(click_y * RAD_SLIDER_MULT)
+
+
+    # New object velocity slider
+    if click_x >= VEL_SLIDER_X_MIN and click_x <= VEL_SLIDER_X_MAX:
+        if (click_y * VEL_SLIDER_MULT != max_vel and
+                click_y * VEL_SLIDER_MULT > 0):
+            max_vel = math.floor(click_y * VEL_SLIDER_MULT)
+
+
+    # Update atoms position
+    for i in range(len(atoms)):
+        atoms[i].x += atoms[i].vel[0]
+        atoms[i].y += atoms[i].vel[1]
+
+        
+    # Make an empty 2D array of all collision sections
+    sections = funcs.sections_build(num_x_sections, num_y_sections)
+
+    # Calculate boundary collisions and allocate atoms to sections
+    # based on position to calculate atom-atom collisions
+    if len(sections) > 1:
+        for i in range(len(atoms)):
+            # Boundary collisions
+            funcs.boundary_collision(atoms[i], boundary)
+
+            # Location coefficient calculation
+            loc_coeffs = funcs.loc_coefficients(atoms[i], boundary,
+                                              base_section_dim, overlap_dim)
+
+            # Allocate atoms to appropriate sections based on location
+            funcs.sections_allocate(atoms[i], sections,
+                                   loc_coeffs[0][0], loc_coeffs[0][1],
+                                   loc_coeffs[0][2], loc_coeffs[1][0],
+                                   loc_coeffs[1][1], loc_coeffs[1][2])
+
+ 
+    # Check for collisions within sub_sections
+    # Does a direct check between all atoms in sub_section
+    for section in sections:
+        for sub_section in section:
+            if len(sub_section) > 1:
+                for a in range(len(sub_section)):
+                    for b in range(a+1, len(sub_section)):
+                        dist = ((sub_section[a].x - sub_section[b].x)**2
+                                +(sub_section[a].y - sub_section[b].y)**2)**0.5
+                        if dist <= sub_section[a].rad + sub_section[b].rad:
+                            funcs.atom_collision(sub_section[a],
+                                                 sub_section[b])
+                                  
+
+    # Control slider interaction
+    # Get x coord only when first clicking, get y coord continuously
+    button = pygame.mouse.get_pressed()
+    if button[0]:
+        pos = pygame.mouse.get_pos()
+        click_y = YDIM - BASE_SLIDER_Y - pos[1]
+        if past_button_state == 0:
+            past_button_state = 1
+            click_x = pos[0]
+    # Reset clicking variables when releasing button
+    elif not button[0] and past_button_state == 1:

```

Comment:
`not button[0]` is guaranteed to be `true` here since this didn't enter the `if button[0]` branch above

Reply:
Wow you went through this way more closely than I expected!